koku/masu/processor/_tasks/data_validation.py

```diff
@@ -0,0 +1,222 @@
+#
+# Copyright 2024 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+import logging
+from datetime import datetime
+
+from django.conf import settings
+
+from api.common import log_json
+from api.provider.models import Provider
+from api.utils import DateHelper
+from masu.database import AWS_CUR_TABLE_MAP
+from masu.database import AZURE_REPORT_TABLE_MAP
+from masu.database import GCP_REPORT_TABLE_MAP
+from masu.database import OCI_CUR_TABLE_MAP
+from masu.database import OCP_REPORT_TABLE_MAP
+from masu.database.report_db_accessor_base import ReportDBAccessorBase
+from masu.util.common import date_range_pair
+from reporting.provider.aws.models import TRINO_LINE_ITEM_DAILY_TABLE as AWS_TRINO_LINE_ITEM_DAILY_TABLE
+from reporting.provider.azure.models import TRINO_LINE_ITEM_DAILY_TABLE as AZURE_TRINO_LINE_ITEM_DAILY_TABLE
+from reporting.provider.gcp.models import TRINO_LINE_ITEM_DAILY_TABLE as GCP_TRINO_LINE_ITEM_DAILY_TABLE
+from reporting.provider.oci.models import TRINO_LINE_ITEM_DAILY_TABLE_MAP as OCI_TRINO_LINE_ITEM_DAILY_TABLE
+
+LOG = logging.getLogger(__name__)
+
+# Filter maps
+TRINO_FILTER_MAP = {
+    Provider.PROVIDER_AWS: {"date": "lineitem_usagestartdate", "metric": "lineitem_unblendedcost"},
+    Provider.PROVIDER_AZURE: {"date": "date", "metric": "coalesce(nullif(costinbillingcurrency, 0), pretaxcost)"},
+    Provider.PROVIDER_GCP: {"date": "usage_start_time", "metric": "cost"},
+    Provider.PROVIDER_OCI: {"date": "lineitem_intervalusagestart", "metric": "cost_mycost"},
+    Provider.PROVIDER_OCP: {
+        "date": "usage_start",
+        "metric": "pod_effective_usage_cpu_core_hours",
+    },
+    "OCPAWS": {"date": "usage_start", "metric": "unblended_cost"},
+    "OCPAzure": {"date": "usage_start", "metric": "pretax_cost"},
+    "OCPGCP": {"date": "usage_start", "metric": "unblended_cost"},
+}
+PG_FILTER_MAP = {
+    Provider.PROVIDER_AWS: {
+        "date": "usage_start",
+        "metric": "unblended_cost",
+    },
+    Provider.PROVIDER_AZURE: {"date": "usage_start", "metric": "pretax_cost"},
+    Provider.PROVIDER_GCP: {
+        "date": "usage_start",
+        "metric": "unblended_cost",
+    },
+    Provider.PROVIDER_OCI: {"date": "usage_start", "metric": "cost"},
+    Provider.PROVIDER_OCP: {
+        "date": "usage_start",
+        "metric": "pod_effective_usage_cpu_core_hours",
+    },
+    "OCPAWS": {"date": "usage_start", "metric": "unblended_cost"},
+    "OCPAzure": {"date": "usage_start", "metric": "pretax_cost"},
+    "OCPGCP": {"date": "usage_start", "metric": "unblended_cost"},
+}
```

Comment:
All bar one of the above metrics are based around standard cost metrics for the providers. The only difference is the ocp one, since this doesn't have cost out of the box, it's set to effective cpu initially.
Additionally we could validate multiple metrics here but initial thoughts are if one metric is incorrect then all probably have a diff. Future iterations we could include blended/amortized and credits etc.